q2_moshpit/busco/tests/test_busco.py

```diff
@@ -5,100 +5,190 @@
 
 
-class TestBUSCO(TestPluginBase):
-    package = "q2_moshpit.busco.tests"
+class MockContext(Mock):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.get_action = Mock()
 
-    @classmethod
-    def setUpClass(self):
 
-        # Set base path
-        p = os.path.join(os.path.dirname(__file__), "data")
+class TestBUSCO(TestPluginBase):
+    package = "q2_moshpit.busco.tests"
 
-        # Get MAGs fixture
+    def setUp(self):
+        super().setUp()
         self.mags = MultiMAGSequencesDirFmt(
-            path=p,
+            path=self.get_data_path('mags'),
             mode="r",
         )
 
-    # Integration test busco.
-    @patch('q2_moshpit.busco.utils._run_busco')
-    @patch('q2_moshpit.busco.utils._zip_busco_plots')
-    @patch('q2_moshpit.busco.utils._draw_busco_plots')
-    @patch('q2_moshpit.busco.utils._collect_summaries_and_save')
-    def test_integration_busco(
-        self,
-        collect_summaries,
-        draw_busco_plots,
-        zip_busco_plots,
-        run_busco
-    ):
-        """
-        Tests entire busco run and patches the previously tested functions.
-        Checks for existence of HTML output.
-
-        Args:
-            collect_summaries (unittest.mock): mock object for function
-                `_collect_summaries_and_save`
-            zip_busco_plots (unittest.mock): mock object for function
-                `_draw_busco_plots`.
-            zip_busco_plots (unittest.mock): mock object for function
-                `_zip_busco_plots`.
-            run_busco (unittest.mock): mock object for function
-                `_run_busco`.
-        """
-        # import shutil
-        # path_to_look_at_html = "/Users/santiago/Downloads/busco_debug_bench"
-
-        with tempfile.TemporaryDirectory() as tmp_path:
-            # This side effect will return the all_summaries_dfs
-            p = self.get_data_path("all_batch_summaries.csv")
-            collect_summaries.return_value = pd.read_csv(p)
-
-            # Run busco
-            evaluate_busco(output_dir=str(tmp_path), bins=self.mags)
-
-            # For render debugging
-            # shutil.copytree(str(tmp_path), path_to_look_at_html)
-
-            # Check for the existence of the html file
-            self.assertTrue(os.path.exists(f"{tmp_path}/index.html"))
-
-            # Assert that the calls where done properly
-            run_busco.assert_called_once_with(
-                output_dir=run_busco.call_args.kwargs['output_dir'],
-                mags=self.mags,
-                params=[
-                    '--mode', 'genome',
-                    '--cpu', '1',
-                    '--contig_break', '10',
-                    '--evalue', '0.001',
-                    '--limit', '3'
-                ]
+    def _prepare_summaries(self):
+        for s in ['1', '2']:
+            os.makedirs(os.path.join(self.temp_dir.name, f"sample{s}"))
```

Comment:
This is a very convenient way of working with temp dirs 😎 . I didn't know `TestPluginBase` objects had this :)